docs/Advanced Configuration/External Database.md

```diff
@@ -0,0 +1,114 @@
+- `username`: The username for the database
+- `password`: The password for the database
+
+If you would like more fine-grained control of the database connection, you can also use the following properties:  
+
+#### Fine-grained Database Configuration
+- `type`: The database type. Available options are `h2` and `postgresql`
+- `hostName`: The host name of the database connection url (e.g. 'localhost')
+- `port`: The port number of the database connection url (e.g. 8080)
+- `name`: The name of the custom database. This should match the name you have set for your database
+
+### Docker Based Configuration
+You can also configure an external database within using Docker. 
+
+#### Adding a Database Container to Docker Compose
+Simply add another service to your Docker Compose file for with the configuration details for the database container. 
+
+```yaml
+services:
+  db:
+    image: 'postgres:17.2-alpine'
+    container_name: db
+    ports:
+      - "5432:5432"
+    environment:
+      POSTGRES_DB: "stirling_pdf"
+      POSTGRES_USER: "admin"
+      POSTGRES_PASSWORD: "stirling"
+```
+
+- `container_name`: This is the name of your database container. This should match the name of the container under `services` as this is what Docker will use to refer to your database
+- `ports`: Specify the port number for your database. The number on the left is the port number the container will access the database internally. The number on the right is the port number the Stirling PDF app will use to connect to the database externally. Ensure this matches the port number in the connection url for your database otherwise the app will not be able to access it.
+- `POSTGRES_DB`: An environment variable for the database container. Specify the name of the custom database here
+- `POSTGRES_USER`: An environment variable for the database container. Specify the username for the database
+- `POSTGRES_PASSWORD`: An environment variable for the database container. Specify the password for the database
+
+You will also need to update the Docker configuration in your app in order to connect to the database: 
+
+```yaml
+services:
+  stirling-pdf:
+    depends_on:
+      - db
+    environment:
+      DOCKER_ENABLE_SECURITY: "true"
+      SYSTEM_DATASOURCE_ENABLECUSTOMDATABASE: "true"
+      SYSTEM_DATASOURCE_CUSTOMDATABASEURL: "jdbc:postgresql://db:5432/stirling_pdf"
+      SYSTEM_DATASOURCE_USERNAME: "admin"
+      SYSTEM_DATASOURCE_PASSWORD: "stirling"
+    # further configuration
+```
+
+- `depends_on`: This specifies any services that your app will need in order to run. Ensure the name matches the container name for your database
+- `DOCKER_ENABLE_SECURITY`: Set this to `true` to enable security features (required for external database configuration)
```

Comment:
You mention this for the docker section but not the yaml section, perhaps a general section before these is needed to say you need security and enterprise etc?

Reply:
Yeah good shout!